Rebrand the app to HeartSMS

Derek52 opened the pull request:
Hi. This took longer than I thought it would because I ran into a few annoying issues. I have it working nicely now though. I left some comments in the form of TODO's for you to look at. They just explain why I made a couple changes, or why I couldn't make a change.
Also, I changed the launcher icon easily. But, in api 26 and later there is a new split icon thing. I didn't touch those, because you need to have the icon in 2 separate png files. One for the foreground, and one for the background. I don't have the skills to separate the foreground from the background in the svg you have here, so I just left the pulse ones in place.
Closes #1

Derek52 committed:
Rename keystore/keys files so I can build the project

Derek52 committed:
Change the package name in the AndroidManifests and in the main build.gradle script.
Change the project name in the main strings.xml file

Derek52 committed:
Refactor package names/paths

Derek52 committed:
Replace pulse icon with heart icon for ic_launchers

Derek52 committed:
Clean up imports and other statements that got messed up by autorenaming packages.

Derek52 committed:
Finally finish errors from import statements not getting renamed properly. The app builds now.

Derek52 committed:
App crashed when I launched settings. It turns out I had even more name changes to fix.

Derek52 committed:
Change primary color to #eb5c72. Changing it to #eb5c72ff causes the app to crash. You can't have a transparent primary color like that. It has to be opaque.

tnyeanderson commented:
@Derek52 thank you very much for the PR! I'm reviewing it now, but it may take me a day or two before merging as there is a lot to inspect.

Derek52 commented:
No problem. Take your time. And I am happy to answer any questions if they come up.

tnyeanderson commented:
Hi @Derek52 , thanks again for your hard work! Looking through this, I see a few issues:

1. Some strings were not renamed properly. For example: `love_pulse_question` in `page-rate-it.xml`. The string was changed in `strings.xml` to `love_Heart_question` (should be `love_heart_question` for consistency), but it was not changed in `page-rate-it.xml`... This is the first thing preventing the app from building, and there are at least a few of these string reference issues. Please update these string names and ensure that they are all lower case and referenced properly. In addition, some `strings.xml` files for other languages were not affected by the rename. I don't expect you to translate every language's strings, but for many of them the word "Pulse" is still present in english, and this should be changed to Heart for those cases (make sure you are doing case-sensitive replace!)
2. For some reason, even after fixing the string references, MessengerApplication.kt is missing references to `xyz.heart.sms.api.implementation` even though I can see the import statements included... not sure what is happening here (again, I'm not an android dev so your help is very much appreciated).
3. The API URLs in `Api.java` were switched back to the ones for pulse somewhere in your commits. These need to be set as they were in 056311d
4. You accidentally moved the `keystore.properties.example` and `api_keys.properties.example` in 0dd9bf04 instead of copying.... please make sure these files get added back to the repo.
5. You can revert the changes made to README.md , I am going to be editing a lot of that and the replacements that were made will lead to dead links.

I'm still making my way through it, but I want to make sure the app can build before I really deep dive.

>But, in api 26 and later there is a new split icon thing. I didn't touch those, because you need to have the icon in 2 separate png files. One for the foreground, and one for the background.

I am looking into this and will try to provide you with the proper files so you can complete this task!

I'm also going to double the bounty because I know this has been a lot of work and I am very thankful for it. Android is way over my head, but I am trying to learn... thank you for your patience :)

Regards,
Thomas

Derek52 committed:
copy keystore and api_keys example files

Derek52 commented:
1. I noticed a couple of those issues. But, I got the app to build and run, so I had assumed I'd fixed them all. I'll go back and make sure they are all gone.

3. I switched the API URL's back to the ones for pulse, because if I change the api url's the app won't function anymore. API urls is something you should change yourself. That's why I reverted them. I din't want to make assumptions about what URL your api would be targeting. 

I will start working on the rest of the fixes right now.

Derek52 committed:
Revert README

Derek52 committed:
Change Pulse to Heart in language strings that use the latin alphabet.
FIx some other naming issues.

Derek52 commented:
I added a few commits that I think take care of everything. Except the app icon of course. If you aren't going to take advantage of the split icon, you can just delete the split pulse icon and use the normal launcher Icons I have already put in the correct places.

Also, I didn't change the API keys. I wanted to make sure you had the server you'd be hitting setup first. Changing the API key's could result in parts of the app not working. If you'd like me to change them back to how I had them in that commit, I can. But, I feel like you should take care of that when you are ready.

Finally, I am not seeing any issues with MessengerApplication.kt. I was on a previous commit, but when I made the PR I had already solved those problems. If you are still seeing issues you should try to do a Build > Clean Project. And maybe a File > Invalidate Cache, and then restart android studio. If you do those things, and are still having problems, just let me know, and I will investigate.

tnyeanderson commented:
Thanks @Derek52 ! The app builds now and appears to work as expected :)

I only really see one other tidbit before I give your work a final review and merge: there are still some colors and styles with "pulse" in the name... 

I'll have some further work to do myself (I figured out how to do the split icon so I can take care of that, as well as changing the readme and support emails, some last few variable names, etc)... but if you could submit one final commit with the color/style names updated I think we will be in business!!

Really appreciate your work on this!!

Derek52 committed:
Change Pulse to Heart in colors and styles xml files.

Derek52 commented:
I believe I found all the colors/styles.

tnyeanderson commented:
@Derek52 you got it! Thanks again! And if you want to take up more bounties, keep your eyes peeled here as I will be opening up some more soon :)

Derek52 commented:
You're welcome. And feel free to tag me on here when you open a new bounty, and I'll happily take a look.

Also, do you know why the bounty hasn't closed on bountysource yet? I haven't actually previously used that website. I found out about this bounty from your reddit listing.

tnyeanderson commented:
@Derek52 yeah I'm not sure why.... I thought it was supposed to close when I merged your PR. You may need to go to the bountysource and click the "Solve Issue" button.  When I click it, it asks "Are you a developer who can solve this issue?" so I think if you go through those steps you may be able to link it up with your merged PR? I'll keep my eye on it if I need to approve anything.

I've only used bountysource a couple of times as a contributor so I don't remember the exact process :( we will get it sorted out though because I want to make sure you get your dough asap!

Derek52 commented:
The solve button just lets you say you've started working on the issue. I've pressed every button I can press on the site. I'll just check the site again later today or tomorrow. Maybe they've got something not working quite right.
